graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,130 @@
+# Alternate Key Pattern
+
+Microsoft Graph API Design Pattern
+
+_The Alternate Key Pattern provides the ability to query for a single, specific resource identifiable through an alternative set of properties that is not its primary key_
+
+## Problem
+
+---
+
+The resources exposed in Graph are identified through a Primary Key - which guarantees uniqueness inside the same resource type. Often though, that same resource can also be uniquely identified by an alternative, more convenient property (or set of properties) that provides a better developer experience.
+
+Take a look at the `user` resource: while the `id` remains a perfectly valid way to get the resource details, the `mail` address is also an unique property that could be used to identify it.
+
+While it is still possible to use the OData filter, such as
```

Comment:
I wonder if we even need to mention OData here.  Can we just say "use the $filter query parameter"